app/src/main/java/com/example/plantsapp/domain/model/Plant.kt

```diff
@@ -1,8 +1,11 @@
 package com.example.plantsapp.domain.model
 
 data class Plant(
-    val name: String,
+    val name: PlantName,
     val speciesName: String,
     val plantPicture: String,
     val wateringFrequencyDays: Int
-)
+) {
+    @JvmInline
+    value class PlantName(val value: String)
```

Comment:
You can name this class right now as Name , not a PlantName

because it is already inside of Plant, you will access it with Plant.Name (and it is more beautiful then Plant.PlantName)